File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSPositionedDeclarationRefImpl.java
package net.kenro.ji.jin.purescript.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import net.kenro.ji.jin.purescript.psi.PSPositionedDeclarationRef;
import net.kenro.ji.jin.purescript.psi.PSVisitor;
import org.jetbrains.annotations.NotNull;

public class PSPositionedDeclarationRefImpl extends PSPsiElement implements PSPositionedDeclarationRef {

    public PSPositionedDeclarationRefImpl(ASTNode node) {
        super(node);
    }

    public void accept(@NotNull PSVisitor visitor) {
        visitor.visitPSPositionedDeclarationRef(this);
    }

    public void accept(@NotNull PsiElementVisitor visitor) {
        if (visitor instanceof PSVisitor) accept((PSVisitor) visitor);
        else super.accept(visitor);
    }





}

File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSPositionedDeclarationRefImpl.kt
package net.kenro.ji.jin.purescript.psi.impl;

import com.intellij.lang.ASTNode;
import net.kenro.ji.jin.purescript.psi.PSPositionedDeclarationRef;

public class PSPositionedDeclarationRefImpl extends PSPsiElement implements PSPositionedDeclarationRef {

    public PSPositionedDeclarationRefImpl(final ASTNode node) {
        super(node);
    }

}
